datafusion/physical-plan/src/windows/mod.rs

```diff
@@ -375,6 +378,16 @@ impl BuiltInWindowFunctionExpr for WindowUDFExpr {
     fn reverse_expr(&self) -> Option<Arc<dyn BuiltInWindowFunctionExpr>> {
         None
     }
+
+    fn get_result_ordering(&self, schema: &SchemaRef) -> Option<PhysicalSortExpr> {
+        self.fun
+            .sort_options()
```

Comment:
The user-defined window function can specify its result ordering which will be used for updating ordering equivalences.